src/precice/config/ParticipantConfiguration.cpp

```diff
@@ -283,7 +274,8 @@ void ParticipantConfiguration::xmlTagCallback(
     mesh::PtrMesh mesh     = _meshConfig->getMesh(meshName);
     PRECICE_CHECK(mesh, "Participant "
                             << "\"" << _participants.back()->getName() << "\" has to use "
-                            << "mesh \"" << meshName << "\" in order to read data from it!");
+                            << "mesh \"" << meshName << "\" in order to read data from it. "
```

Comment:
Somehow this condition seems to not be reached. If I try to **read** from a mesh that is not used, I still get the error message for **writing**.

For example, in the C++ solver dummy:
```xml
    <participant name="SolverOne">
      <use-mesh name="MeshOne" provide="yes"/>
      <write-data name="Forces"     mesh="MeshOne" />
      <read-data  name="Velocities" mesh="MeshTwo" />
    </participant>
```
gives (before, similarly now):
```
preCICE:ERROR:  Participant "SolverOne" has to use mesh "MeshTwo" when writing data to it!
```

Reply:
That's due to the check in line 475 (new numbering), see below, I fixed it there.
This check here only fires if the mesh does not exist at all. So, the message is not wrong, but neither perfect.

Reply:
Good point, I did not notice that the previous one was already wrong and simply rephrased it. Thanks!